pkg/server/router/presentation.go

```diff
@@ -202,7 +240,24 @@ type OperationResult struct {
 // @Failure      500      {string}  string  "Internal server error"
 // @Router       /v1/presentations/submissions [put]
 func (pr PresentationRouter) CreateSubmission(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
-	var resp Operation
+	var request CreateSubmissionRequest
+	if err := framework.Decode(r, &request); err != nil {
+		return framework.NewRequestError(util.LoggingErrorMsg(err, "invalid create submission request"), http.StatusBadRequest)
```

Comment:
may be easier to read in two lines (util.Logging separate from the return)

Reply:
Done